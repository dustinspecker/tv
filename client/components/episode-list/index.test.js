import EpisodeList from '.'
import React from 'react'
import renderer from 'react-test-renderer'

test('EpisodeList displays list of episodes', () => {
  fetch.mockResponseOnce(
    JSON.stringify({
      shows: [
        {name: 'Episode 1', episode: '/show/season1/episode1'},
        {name: 'Episode 2', episode: '/show/season1/episode2'}
      ]
    })
  )

  const props = {
    match: {
      params: {
        season: 'seasonname',
        showname: 'showname'
      }
    }
  }

  const element = <EpisodeList {...props} />

  const component = renderer.create(element)
  expect(component.toJSON()).toMatchSnapshot()

  return new Promise(resolve => setTimeout(resolve, 0)).then(() => {
    component.update(element)
    expect(component.toJSON()).toMatchSnapshot()
    expect(fetch.mock.calls.length).toEqual(1)
    expect(fetch.mock.calls[0][0]).toEqual('/tv/showname/seasonname')
  })
})
