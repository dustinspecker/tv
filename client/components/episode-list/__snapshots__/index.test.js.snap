// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeList displays list of episodes 1`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
/>
`;

exports[`EpisodeList displays list of episodes 2`] = `
<ul
  className="MuiList-root-1 MuiList-padding-2"
>
  <div
    className="MuiButtonBase-root-17 MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 MuiListItem-button-14"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiListItemText-root-20"
    >
      <span
        className="MuiTypography-root-26 MuiTypography-subheading-33 MuiListItemText-primary-23"
      >
        Episode 1
      </span>
    </div>
    <span
      className="MuiTouchRipple-root-61"
    />
  </div>
  <div
    className="MuiButtonBase-root-17 MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 MuiListItem-button-14"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiListItemText-root-20"
    >
      <span
        className="MuiTypography-root-26 MuiTypography-subheading-33 MuiListItemText-primary-23"
      >
        Episode 2
      </span>
    </div>
    <span
      className="MuiTouchRipple-root-61"
    />
  </div>
</ul>
`;
